word_guess.rb

```diff
@@ -0,0 +1,325 @@
+    puts "==================================================================\n"
+    print "                            " #to move the word blanks away from left justified
+    @answer_validation_array.each do |letter|
+      if @guessed_things.include? letter
+        print " #{letter} "
+      else
+        print " * "
+      end
+    end
+    puts "\n==================================================================\n"
+  end
+
+  def display_guessed_letters
+    if !(@guessed_things.empty?)
+      print "\nRemember that you have already guessed "
+      print "#{@guessed_things}"
+      print "."
+    end
+  end
+
+  def ask_user_for_guess
+    @last_guess = @user_guess.user_prompt
+    if @last_guess.length > 1
+      @last_guess_array = @last_guess.split(//)
+    end
+  end
+
+  def guessed_word?
+    @last_guess.length > 1
+
+  end
+
+  def add_to_guess_array
+    @guessed_things << @last_guess
+    @guessed_things.sort! #because we will display guessed letters in a reasonable fashion
+  end
+
+  def update_remaining_guesses #need to test - writing test below
+    if !(@answer_validation_array.include? @last_guess) #if the guess is not in the answer array, the user losses a guess.
+      @number_of_guesses -= 1
+      puts "Oh no!  There was no '#{ @last_guess }' in the word."
+      puts "The dinosaur is now #{ @number_of_guesses } feet away from you!!!!"
+    else
+      puts "Yipee!!! '#{ @last_guess }' was in the word."
+      puts "The dinosaur is still #{ @number_of_guesses } feet away."
+    end
+  end
+
+
+end
+
+class GuessPrompter
+
+  def initialize
+    @guessed_things = []
+  end
+
+  def user_prompt
+    @possible_letters = ('a'..'z').to_a
+    print "\nWhat letter would you like to guess? Type '*' if you would like to guess the whole word. > "
+    #this loop validates that the user input is a SINGLE LETTER.  If it is a single letter, the second if statement checks that the user has not alreay guessed that letter by seeing what's included in the guessed_letters array.
+    while @guess = gets.chomp.downcase
+      if @guess == "*"
+       print  "What is the word? > "
+        return @whole_word_guess = gets.chomp
+        break
+
+      elsif !(@possible_letters.include? @guess) || !(@guess.length == 1) # first we need to get a letter
+        print "Sorry, that is not a valid guess. Please enter a letter.  > "
+      else
+        if @guessed_things.include? @guess #guessed letters IS part of the array already
+          print "You've already guessed that letter!  Please enter a new letter. > "
+        else
+          @guessed_things << @guess #push the guess into the array for future comparisons
+          return @guess
+          break
+        end
+      end
+    end
+  end
+
+end
+
+#include bank of words, get_word method that will choose a word
+class WordFetcher
+
+  def initialize
+    @words = %w(cat dog hello there pie mug hat fox door bear frog mice)
+
+  end
+
+  def get_word
+    @answer = @words.sample
```

Comment:
Since the `@answer` variable is only used within this method, it would make sense to use a local variable (without the `@`). This is because `@answer` is not used anywhere outside of this single method.